packages/app/src/cli/services/function/build.ts

```diff
@@ -12,7 +12,86 @@ import {AbortSignal} from '@shopify/cli-kit/node/abort'
 import {renderTasks} from '@shopify/cli-kit/node/ui'
 import {pickBy} from '@shopify/cli-kit/common/object'
 import {runWithTimer} from '@shopify/cli-kit/node/metadata'
-import {Writable} from 'stream'
+import cachedir from 'cachedir'
```

Comment:
We already have a dependency for this, is called `env-paths`.

It's available only in `cli-kit` but you can either re-export the cache methods from there.